### ISSUE TITLE
Fatal NodeJS Error w/ Climate Sensor

### ISSUE THREAD

**Issue by sinkr** [friedemannf/homebridge-mqtt-eve#1]
Thanks for the plugin.

Homebridge 1.6.1, Ubuntu 22.04.2, Node 18.6.1.

Maybe I am not configuring the plugin correctly (it's not really clear what I should be putting in `mqtt_topic`).

Thanks in advance for your help!

Config:
```
        {
            "name": "Basement Eve",
            "type": "climate",
            "mqtt_topic": "zigbee2mqtt/device1",
            "serial_number": "<real-sn-redacted>",
            "history": {
                "storage": "fs"
            },
            "accessory": "mqtt-eve-accessory"
        }
```

Error:
```
10/8/2023, 9:42:19 AM] [Basement Eve] Initializing mqtt-eve-accessory accessory...
[10/8/2023, 9:42:19 AM] TypeError: Cannot read properties of undefined (reading 'substring')
    at Object._createHistoryService (/usr/lib/node_modules/homebridge-mqtt-eve/src/Accessory.js:194:27)
    at new <anonymous> (/usr/lib/node_modules/homebridge-mqtt-eve/src/Accessory.js:89:33)
    at /usr/lib/node_modules/homebridge/src/server.ts:379:50
    at Array.forEach (<anonymous>)
    at Server.loadAccessories (/usr/lib/node_modules/homebridge/src/server.ts:299:29)
    at Server.start (/usr/lib/node_modules/homebridge/src/server.ts:179:12)
[10/8/2023, 9:42:19 AM] Got SIGTERM, shutting down Homebridge...
[10/8/2023, 9:42:19 AM] AssertionError [ERR_ASSERTION]: Cannot generate setupURI on an accessory that isn't published yet!
    at Bridge.Accessory.setupURI (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Accessory.ts:994:11)
    at Server.setServerStatus (/usr/lib/node_modules/homebridge/src/server.ts:155:45)
    at Server.teardown (/usr/lib/node_modules/homebridge/src/server.ts:199:10)
    at signalHandler (/usr/lib/node_modules/homebridge/src/cli.ts:93:12)
    at process.emit (node:events:513:28)
    at process.emit (/usr/lib/node_modules/homebridge/node_modules/source-map-support/source-map-support.js:516:21)
```

**Comment by friedemannf** [comment]
Hey there!

The `mqtt_topic` should be set to the topic that the device will push updates on. If you're using zigbee2mqtt this will by default be `zigbee2mqtt/<FRIENDLY_NAME>`. If you've configured your device as `device1` in z2m your topic should be fine, if not you'd have to adjust that value. I'd expect most users to use this plugin in combination with z2m, that's why the default value already includes the prefix.

The error you're seeing is unrelated to that and in fact a bug. It should normally set the name of the file where it stores the history automatically based on your serial number, but seems like this is broken. I'll take a look at this and will publish a new version that will address this error - thanks for reporting!

In the meantime you can try setting the filename manually. As an example, this is the config I'm using for one of my devices:
```
        {
            "name": "Bathroom Climate",
            "type": "climate",
            "mqtt_url": "mqtt://localhost:1883",
            "mqtt_topic": "zigbee2mqtt/bathroom_wall_thermostat",
            "serial_number": "bathroom_wall_thermostat",
            "history": {
                "storage": "fs",
                "filename": "bathroom_wall_thermostat.json"
            },
            "keys": {
                "temperature": "local_temperature"
            },
            "accessory": "mqtt-eve-accessory"
        }
```
The `keys` section is optional and only needs to be set if your device uses json keys other than `temperature`, `humidity`, `pressure`. But from your config it looks like you're missing the MQTT URL. I'll also work on the documentation and will add a few examples.

Let me know if that works for you, I'm happy to help!

**Comment by sinkr** [comment]
Hey, thank you for the response!   `"filename":` was the fix for the crash. 

I may have misunderstood the plugin's intent, because my Eve devices show up in HomeKit already.  I was hoping that this plugin would allow me to see the historical data in HomeKit w/o having to revert to the Eve app which is a nightmare.  Is this the case, or is it just the JSON file on disk?

With this in mind, I have 2 questions:

1. Where is the JSON file stored?  I don't see it in `/var/lib/homekit/persist`, and
2. why is the reading from your plugin roughly half what the Eve is reading?  Presently Eve/HomeKit reads 63F, while Eve/homebridge-mqtt-eve reads 32F.

Again, thanks for your help--regards,

-R.

**Comment by friedemannf** [comment]
Awesome we got that sorted out!

Unfortunately, yes. This plugin is actually doing the opposite: taking data via MQTT and emulating Eve devices.
The idea is to use any arbitrary sensor and add the history functionality in the Eve.app to it. This could look as follows:
```
SONOFF SNZB-02 -> Zigbee2mqtt -> MQTT -> Homebridge -> This plugin -> Eve.app
```
It sounds like this is different from what you're looking for. Eve uses a proprietary format to enable the history graphs in their app, it's therefore impossible to see those in Apple's Home app.

On your questions:

1. The JSON file will by default be stored in the homebridge base directory. But it will only be created once there's actually data. Normally that should be `~/.homebrige`.

2. When unconfigured the plugin reports a value of 0°C or 32°F, this would change once data actually starts coming in.

Let me know if there's anything else you need!

**Comment by friedemannf** [comment]
I've published a new version, 1.1.2, that addresses the issue with the filename causing a crash. I'll close this issue for now - if there's anything else you'd like do see, feel free to let me know!
